hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -317,10 +317,20 @@ def __repr__(self):
         bits = []
         for name, setting in all_settings.items():
             value = getattr(self, name)
+            # The only settings that are not shown are those that are
+            # deprecated and left at their default values.
             if value != setting.default or not setting.hide_repr:
                 bits.append('%s=%r' % (name, value))
         return 'settings(%s)' % ', '.join(sorted(bits))
 
+    def show_changed(self):
+        bits = []
+        for name, setting in all_settings.items():
+            value = getattr(self, name)
+            if value != setting.default or not setting.hide_repr:
+                bits.append('%s=%r' % (name, value))
+        return ' '.join(sorted(bits))
```

Comment:
IMO `', '.join(sorted(bits, key=len))` gives somewhat nicer output - mostly by forcing the database to last - but up to you here.  I do think the comma helps distinguish the parts though.

Reply:
Just copied the formatting from pytest-benchmark. But upon further research there doesn't appear to be any standard on how to display the settings in pytest plugins so I think we are fine to do what we think is best.